cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -318,7 +323,12 @@ public boolean scanApplications(ScanApplicationsRequest request,
    */
   public boolean scanSourceControlMetadata(ScanSourceControlMetadataRequest request,
       int txBatchSize,
-      Consumer<SourceControlMetadataRecord> consumer) throws IOException {
+      Consumer<SourceControlMetadataRecord> consumer)
+      throws IOException, RepositoryNotFoundException {
+    NamespaceId namespaceId = new NamespaceId(request.getNamespace());
+    // Checking if repository config is present in the namespace
+    sourceControlMetadataRefresher.getRepositoryMeta(namespaceId);
```

Comment:
What's the point of calling get without using the result?

Reply:
We want to check if RepositoryConfig exist or throw execption otherwise.